utils/file-tester/lms_file_tester/__main__.py

```diff
@@ -0,0 +1,152 @@
+        format=logFormatter,
+        level=level,
+    )
+
+
+def _report(errors: List[str], happy: str):
+    if len(errors) == 0:
+        logger.info(happy)
+    else:
+        for e in errors:
+            logger.warning(e)
+
+
+def _validate_users_and_sections(input_directory: str):
+    errors = validate_base_directory_structure(input_directory)
+    _report(errors, "Basic directory structure is valid.")
+
+    errors = validate_users_file(input_directory)
+    _report(errors, "Users file is valid.")
+
+    errors = validate_sections_file(input_directory)
+    _report(errors, "Sections file is valid.")
+
+
+def _validate_assignment_directories(input_directory: str, section_id: Union[int, str]):
+    assignments = fread.get_assignments(input_directory, section_id)
+
+    if assignments.empty:
+        # This just means there is no file in the given directory
+        return
+
+    for _, assignment_id in assignments["SourceSystemIdentifier"].iteritems():
+        errors = validate_assignment_directory_structure(
+            input_directory, section_id, assignment_id
+        )
+        _report(
+            errors,
+            f"Assignment directory structure is valid for section {section_id}, assignment {assignment_id}.",
+        )
+
+    errors = validate_submissions_file(input_directory, assignments)
+    _report(errors, "Submissions file is valid.")
+
+
+def _validate_section_directories(input_directory: str):
+    sections = fread.get_all_sections(input_directory)
+
+    for _, section_id in sections["SourceSystemIdentifier"].iteritems():
+        errors = validate_section_directory_structure(input_directory, section_id)
+        _report(
+            errors, f"Section directory structure is valid for section {section_id}"
+        )
+
+        _validate_assignment_directories(input_directory, section_id)
+
+    errors = validate_attendance_events_file(input_directory, sections)
+    _report(errors, "Attendance Events file is valid.")
+
+    errors = validate_grades_file(input_directory, sections)
+    _report(errors, "Grades file is valid.")
+
+    errors = validate_assignments_file(input_directory, sections)
+    _report(errors, "Assignments file is valid.")
+
+    errors = validate_section_associations_file(input_directory, sections)
+    _report(errors, "Section Associations Events file is valid.")
+
+    errors = validate_section_activities_file(input_directory, sections)
+    _report(errors, "Section Activities Events file is valid.")
+
+
+def _validate_system_activities_directories(input_directory: str):
+    errors = validate_system_activities_directory_structure(input_directory)
+    _report(errors, "System Activities directory structure is valid.")
+
+    errors = validate_system_activities_file(input_directory)
+    _report(errors, "System Activities file is valid.")
+
+
+def _main():
+
+    if len(sys.argv) != 2:
+        logger.critical(
+            "Must pass an input directory as the only argument to the script."

```

Comment:
I think this validation should be mentioned in a readme file.

Reply:
Done